### PR TITLE
feat(api): add POST /api/v1/credentials/verify endpoint

### PR DESCRIPTION
**Pull request by ashleythedeveloper**
This PR adds an unauthenticated credential verification API route that fetches a verifiable credential from a storage URI, optionally decrypts it, checks its integrity hash, and verifies the credential signature via the system VC service. This extracts server-side verification logic from the client-side verify page, preparing for a follow-up PR that refactors the page to consume this API.

## Changes

- **`withPublicRoute` wrapper** (`src/lib/api/with-public-route.ts`): lightweight route wrapper mirroring `withTenantAuth` without auth; provides correlation IDs, request context, duration logging, and error handling
- **Middleware update** (`src/middleware.ts`): `PUBLIC_API_ROUTES` allowlist so the verify endpoint bypasses auth while still receiving correlation IDs and header stripping
- **Route handler** (`src/app/api/v1/credentials/verify/route.ts`): 8-step pipeline with step-level logging, encryption detection via `isEncryptedEnvelope`, hash verification, credential type validation (array and string), VC service verification, JWT decoding with shape validation, and a `warnings` array for non-fatal issues
- **`isEncryptedEnvelope` utility** (`packages/services/src/encryption/is-encrypted-envelope.ts`): reusable encrypted envelope detection extracted to the services package
- **E2E tests** (`e2e/cypress/e2e/credential_api/credential-verify.cy.ts`): full integration tests covering happy paths (encrypted and unencrypted), validation errors, processing errors, and unauthenticated access

## Test plan
- [x] 30 route handler unit tests (input validation, upstream errors, credential processing, verification outcomes, edge cases, warnings)
- [x] 10 withPublicRoute wrapper tests
- [x] 8 middleware tests (public route bypass, auth enforcement, header stripping)
- [x] 5 isEncryptedEnvelope utility tests
- [x] E2E tests: verify unencrypted credential, verify encrypted credential, unauthenticated access, validation errors (missing uri, invalid format, bad scheme, bad hash), processing errors (missing key, hash mismatch, unreachable URI)
- [x] Lint clean on all new files

### CODE REVIEW

File: packages/reference-implementation/src/app/api/v1/credentials/verify/route.ts [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+  ) {
+    throw new ValidationError('decryptionKey must be a 64-character hex string');
+  }
+
+  // ── Step 2: Fetch credential from storage URI ──────────────────────
+  logger.info({ uri: body.uri }, 'Fetching credential from storage');
+
+  let fetchResponse: Response;
+  try {
+    fetchResponse = await fetch(body.uri, { signal: AbortSignal.timeout(10_000) });
```
[the remark this answers is not in the document]
Comment (ashleythedeveloper) [reply packages/reference-implementation/src/app/api/v1/credentials/verify/route.ts:179]: Addressed in 313c352. Protocol restriction (HTTP/HTTPS only) is enforced via URI scheme validation. A domain allowlist is not appropriate here as this is a public verification endpoint that must accept credentials stored at arbitrary URIs. For production deployments, SSRF protection against private IP ranges should be handled at the infrastructure level (reverse proxy/API gateway), as noted in the code comment.

File: packages/reference-implementation/src/app/api/v1/credentials/verify/route.ts [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+
+  if (!fetchResponse.ok) {
+    const message = `Failed to fetch credential: storage returned ${fetchResponse.status}`;
+    logger.warn({ uri: body.uri, status: fetchResponse.status }, 'Credential fetch failed');
+    return NextResponse.json({ error: message, code: 'UPSTREAM_ERROR' }, { status: 502 });
+  }
+
+  let responseText: string;
+  try {
+    responseText = await fetchResponse.text();
```
[the remark this answers is not in the document]
Comment (ashleythedeveloper) [reply packages/reference-implementation/src/app/api/v1/credentials/verify/route.ts:197]: Acknowledged. The 10MB configurable limit (VERIFY_MAX_CREDENTIAL_SIZE) provides a reasonable upper bound. Switching to streaming would require significant refactoring (chunked reading with back-pressure) for a marginal improvement — credentials in practice are typically < 100KB. The 10-second fetch timeout also limits exposure. If this becomes a concern, it can be addressed in a follow-up.

File: packages/reference-implementation/src/app/api/v1/credentials/verify/route.ts [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+    if (!body.decryptionKey) {
+      logger.info('Encrypted credential but no decryption key provided');
+      return NextResponse.json(
+        { error: 'Credential is encrypted but no decryptionKey was provided', code: 'DECRYPTION_REQUIRED' },
+        { status: 422 },
+      );
+    }
+
+    logger.info('Decrypting credential');
+    const encrypted = fetchedData as { cipherText: string; iv: string; tag: string; type: string };
```
[the remark this answers is not in the document]
Comment (ashleythedeveloper) [reply packages/reference-implementation/src/app/api/v1/credentials/verify/route.ts]: Fixed in 313c352. Now using `isEncryptedEnvelope(fetchedData)` directly in the `if` condition so TypeScript narrows the type automatically — no explicit cast needed.

File: packages/reference-implementation/src/app/api/v1/credentials/verify/route.ts [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+  const credentialId = credential.id;
+  if (typeof credentialId !== 'string') {
+    logger.warn({ credentialIdType: typeof credentialId }, 'Credential id is not a string');
+    warnings.push('Credential id is not a string; unable to decode JWT');
+  } else if (!credentialId.startsWith('data:application/vc+jwt,')) {
+    logger.warn('Credential id does not use the expected data:application/vc+jwt media type');
+    warnings.push('Credential id does not use the expected data:application/vc+jwt media type');
+  } else {
+    try {
+      const jwt = credentialId.replace('data:application/vc+jwt,', '');
```
[the remark this answers is not in the document]
Comment (ashleythedeveloper) [reply packages/reference-implementation/src/app/api/v1/credentials/verify/route.ts]: Fixed in 313c352. Extracted `JWT_PREFIX` constant and using `substring(JWT_PREFIX.length)` instead of `replace()`.
